src/icrc1-agent.mo

```diff
@@ -35,13 +36,7 @@ module {
 
   public type FeeResult = R.Result<Nat, { #CallIcrc1LedgerError }>;
 
-  public class LedgerAgent(api : ICRC1.API) {
-    var fee_ = 0;
-
-    public func fee() : Nat = fee_;
-
-    public func setFee(x : Nat) = fee_ := x;
-
+  public class LedgerAgent(api : ICRC1.API, feeManager : FeeManager.FeeManager) {
```

Comment:
This is bad. I wanted to make the low-level layers icrc1-agent, icrc84-helper independent from this application (independent from the token handler). They should be publishable by themselves later, stand-alone. They can't depend on FeeManager.

Reply:
Some background: In my previous designs the higher level (=DepositManager) was setting the fee in the lower level (=icrc1-agent) and the lower level was only using the set fee, never changing it. When BadFee happens the lower level only informs the higher level but does not adapt its own fee.

The reason for that is more generality. The ICRC-1 standard does not specify what happens if try to make a transfer with a too high fee. Current implementations reject that. They require the fee to exactly match. But that is not specified in the ICRC-1 spec. There could be ledger that accept a higher fee (and maybe prioritise those like in Bitcoin, Ethereum, etc.). In my model the higher level could run the lower level at a higher than necessary fee. And when the lower level detects a BadFee then the higher level can decide what to update the fee to. It is just a mental model that is more general.

A more plausible scenario than prioritization based on fee is a scenario where the ICRC-1 ledger has a fluctuating fee based on demand. And the higher level code does not want to track the fluctuation. It just wants to set a consistently higher fee with some safety margin so that its transfer rarely fail. Then it would set the fee to a value higher than the minimum.